File: Garage_Rental.Infra/Service/PaymentService.cs
using Garage_Rental.Core.Data;
using Garage_Rental.Core.Repository;
using Garage_Rental.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

namespace Garage_Rental.Infra.Service
{
    public class PaymentService : IGenericService<Payment>
    {
        private readonly IGenericRepository<Payment> _paymentRepository;

        public PaymentService(IGenericRepository<Payment> PaymentRepository)
        {
            _paymentRepository = PaymentRepository;
        }

        public bool Create(Payment t)
        {
            return _paymentRepository.Create(t);
        }

        public void Delete(int id)
        {
             _paymentRepository.Delete(id);
        }

        public List<Payment> GetAll()
        {
            return _paymentRepository.GetAll();
        }

        public Payment GetById(int id)
        {
            return _paymentRepository.GetById(id);
        }

        public void Update(Payment t)
        {
              _paymentRepository.Update(t);
        }
    }
}

File: FrontEnd/GarageRentalFrontEnd/src/app/Services/payment.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Payment } from '../Models/payment.model';

@Injectable({
  providedIn: 'root'
})
export class PaymentService {


 baseApiUrl:string = environment.baseApiUrl;

  constructor(private http:HttpClient) { }


  getPaymentList(): Observable<Payment[]>{
    return this.http.get<Payment[]>(this.baseApiUrl +'Payment/GetAll');
  }

  getPayment(id:string):Observable<Payment>{
    return this.http.get<Payment>(this.baseApiUrl +'Payment/GetById?id=' + id);
  }

  addPayment(addPaymentReq:Payment):Observable<Payment>{
    return this.http.post<Payment>(this.baseApiUrl +'Payment/Create',addPaymentReq);
  }

  updatePayment(id:number,updatePaymentreq:Payment):Observable<Payment>{
    return this.http.post<Payment>(this.baseApiUrl +'Payment/Update?id='+ id ,updatePaymentreq)
  }
  deletePayment(id:number):Observable<Payment>{
    return this.http.delete<Payment>(this.baseApiUrl +'Payment/Delete?id=' + id);
  }
}
